packages/custom-storage/src/sql/SQLStorage.ts

```diff
@@ -350,6 +290,120 @@ export class SQLStorage {
     return result.rows[0];
   }
 
+  private async preventAccidentalDatasetsOverwrite(overwriteDatasets: boolean, datasets: Array<Dataset|string>) {
+    const fullDatasets = datasets.filter((dataset): dataset is Dataset => typeof dataset !== 'string');
+    const existingTables = await this.checkExistingDataset(fullDatasets);
+    if (!overwriteDatasets) {
+      if (existingTables.length > 0) {
+        throw new DuplicatedDatasetsError(existingTables.map((dataset) => dataset.name));
+      }
+    }
+  }
+
+  private async insertVisTable(vis: Visualization) {
+     const insertResult: any = await this._sql.query(`INSERT INTO ${this._tableName}
+     (${this.FIELD_NAMES_INSERT.join(', ')})
+     VALUES
+     (
+       ${
+         this.FIELD_NAMES_INSERT
+           .map((field: string) => {
+             const visField = this.VIS_FIELDS[field];
+             const fieldValue = (vis as any)[field];
+
+             const value = visField && visField.format ? visField.format(fieldValue) : fieldValue;
+             return value === null ? 'null' : value;
+           })
+           .join()
+       }
+     )
+     RETURNING id, lastmodified
+   `);
+
+     if (insertResult.error) {
+      return null;
+    }
+
+     const insertedVis = insertResult.rows[0];
+     return {
+      id: insertedVis.id,
+      lastModified: insertedVis.lastmodified
+    };
+  }
+
+  private async updateVisTable(vis: StoredVisualization) {
+    const updatedResult: any = await this._sql.query(`UPDATE ${this._tableName}
+    SET
+      ${
+        this.FIELD_NAMES_INSERT
+          .map((field: string) => {
+            const visField = this.VIS_FIELDS[field];
+            const fieldValue = (vis as any)[field];
+
+            const value = visField && visField.format ? visField.format(fieldValue) : fieldValue;
+            return `${field} = ${value === null ? 'null' : value}`;
+          })
+          .join()
+      }
+      ,${this.VIS_FIELDS.lastModified.name}=NOW()
+    WHERE ${this.VIS_FIELDS.id.name} = '${vis.id}'
+    RETURNING ${this.VIS_FIELDS.id.name}, ${this.VIS_FIELDS.lastModified.name}
+  `);
+
+    if (updatedResult.error) {
+     return null;
+   }
+
+    return {
+     id: vis.id,
+     lastModified: updatedResult.rows.length ? updatedResult.rows[0].lastmodified : vis.lastModified
```

Comment:
I modified your `rows[0].lastModified` with `.lastmodified`. Double check it please

Reply:
It is right